src/battle_script_commands.c

```diff
@@ -8661,39 +8661,27 @@ static void Cmd_various(void)
         i = TRUE;
         if (gBattleMons[gBattlerAttacker].status1 & STATUS1_PARALYSIS)
         {
-            if (targetAbility == ABILITY_LIMBER)
+            if (!(CanBeParalyzed(gBattlerTarget)))
             {
                 gBattlerAbility = gBattlerTarget;
                 BattleScriptPush(T1_READ_PTR(gBattlescriptCurrInstr + 3));
                 gBattlescriptCurrInstr = BattleScript_PRLZPrevention;
                 i = FALSE;
             }
-            else if (IS_BATTLER_OF_TYPE(gBattlerTarget, TYPE_ELECTRIC))
-            {
-                BattleScriptPush(T1_READ_PTR(gBattlescriptCurrInstr + 3));
-                gBattlescriptCurrInstr = BattleScript_PRLZPrevention;
-                i = FALSE;
-            }
             else
             {
                 gBattleCommunication[MULTISTRING_CHOOSER] = 3;
             }
         }
         else if (gBattleMons[gBattlerAttacker].status1 & STATUS1_PSN_ANY)
         {
-            if (targetAbility == ABILITY_IMMUNITY)
+            if (!CanBePoisoned(gBattlerTarget, targetAbility))
```

Comment:
Shouldn't this be `if  (!CanBePoisoned(gBattlerAttacker, gBattlerTarget))`? That's what its arguments say.